packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1050,6 +1060,23 @@ export class ContainerRuntime
 		return runtime;
 	}
 
+	// Promise to load runtime attributor after loading module.
+	private runtimeAttributorP: Promise<IRuntimeAttributor> | undefined;
+	private _runtimeAttributor: IRuntimeAttributor | undefined;
+	// Represents whether the socket module is loaded or not.
+	private runtimeAttributorModuleLoaded: boolean = false;
+
+	public get IRuntimeAttributor(): IRuntimeAttributor {
+		if (this._runtimeAttributor === undefined) {
+			throw new UsageError("Attribution not available");
+		}
+		return this._runtimeAttributor;
+	}
+
+	private async runtimeAttributorLoadPromise(): Promise<IRuntimeAttributor | undefined> {
```

Comment:
Given it's used only for finishing load, maybe consider returning Promise<void> and renaming to reflect such change